logger.go

```diff
@@ -79,3 +78,45 @@ func Sublogger(inp ZapCompatibleLogger, subname string) (loggerRet ZapCompatible
 
 	return loggerRet
 }
+
+// LogWith adds fields for logging to a given ZapCompatibleLogger instance.
+// This function uses reflection to dynamically add fields to the provided logger by
+// calling its `WithFields` method if it is an RDK logger, or its `With` method if it is a Zap logger.
+// If neither method is available, it logs a debug message and returns the original logger.
+func LogWith(inp ZapCompatibleLogger, args ...interface{}) (loggerRet ZapCompatibleLogger) {
+	loggerRet = inp
+
+	defer func() {
+		if r := recover(); r != nil {
+			inp.Debugf("panic occurred while adding fields to logger: %v, returning self", r)
+		}
+	}()
+
+	reflectArgs := make([]reflect.Value, len(args)+1)
+	reflectArgs[0] = reflect.ValueOf(inp)
```

Comment:
I think a small word about the convention here would be helpful. Specifically:
```
When using reflection to call receiver methods, the first argument must be the object. The remaining arguments are the actual function parameters.
```

Apologies if that critique have been applied to the prior CR and I'm being inconsistent.